arednNetworkDesign/channel_planning.rst

```diff
@@ -0,0 +1,106 @@
+================
+Channel Planning
+================
+
+The previous section identified the different channels in each frequency band which are available for AREDN |trade| networking. Devices on each side of a radio link must use the same frequency band, channel, and channel width. Beyond that requirement, however, you have quite a bit of freedom to select the radio channels that will ensure the highest signal quality for your network.
+
+In a basic AREDN |trade| *meshnet* with individual nodes spread across a limited geographical area, all of the nodes may use the same frequency, channel, and channel width. This allows them to establish a network and route data to any of the sites as needed. However, as more nodes join the network or when several nodes are :abbr:`collocated (sharing the same physical site)`, it is possible for them to interfere with each other. In order for an AREDN |trade| network to scale up in size and complexity, channel planning becomes increasingly important.
+
+Some amount of data traffic is required for :abbr:`OLSR (Optimized Link State Routing protocol)` to maintain and operate the mesh network. Having a growing number of nodes on the same channel will increase the amount of :abbr:`OLSR (Optimized Link State Routing protocol)` handshaking, which can lead to more latency for data traffic across the network. Any application, program, or service that is sensitive to latency (especially voice or video services) may experience difficulty or even become unusable.
+
+Another case is when there is one poor quality link over which all traffic must be routed. The handshaking and data retransmissions may cause all the other links to wait. The entire network can be impacted by one low quality path which becomes a single bottleneck. If at all possible you should increase the signal quality of that vital link, or establish a better link as an alternate path.
+
+The Hidden Node Problem
+-----------------------
+
+In any wireless network there will be nodes which are not within radio range of each other. In the example below, **A** can hear **B** but cannot hear **C**. Since **A** and **C** are hidden from each other, they may try to transmit on the same channel at the same time without knowing it. *Collision Detection* mechanisms will not help because the nodes have no way to communicate except through node **B**.
+
+.. image:: _images/hidden-node.png
+   :alt: Hidden Node Problem
+   :align: center
+
+AREDN |trade| firmware uses IEEE 802.11 :abbr:`RTS (Request to Send)`/:abbr:`CTS (Clear to Send)` messages and *Collision Avoidance* mechanisms help solve this problem. When a node wants to transmit, it sends a :abbr:`RTS (Request to Send)` message and waits until it receives the corresponding :abbr:`CTS (Clear to Send)` message. Decreased network throughput is one side effect of this scheme because several nodes may be negotiating or waiting to use the channel.
```

Comment:
CTS/RTS is a technique on top of CSMA (which can be called 'collision avoidance').  Technically, in 802.11 the node does a Clear Channel Assessment (CCA) to listen if the channel is available.  It then goes ahead and transmits accordingly.    RTS/CTS is not mandated and not used by all traffic.      CTS/RTS is only done for large packets, don't recall the exact size, but probably max size packets only.   I'm not entirely sure the full list, but thinks like broadcast packets, management frames, and some other things don't use RTS/CTS.

Reply:
Cool.  Maybe I should "genericize" the messaging description -- just say there is channel checking that happens.  Or do you think we should include more in-depth details on the exact types of messaging transactions?  Don't want to bury people with details unless we're expecting they have the background/expertise to understand them.  What are your thoughts?

Reply:
I'd recommend using just-enough descriptions, but use the formal term, then reference material where they can find and drill down into the detail should they desire.  I'm a big proponent of not copying or re-inventing a wheel, unless there's cause to do so, e.g.  we want a self contained copy.    Copy enough to have just-enough information the reader can get a conceptual understanding, then refs to dive into detail.  

For example,  "AREDN uses IEEE 802.11a/b/g/n"  To explain all the detail reference:

https://en.wikipedia.org/wiki/IEEE_802.11n-2009

"802.11n uses CSMA to determine when to transmit on the RF channel"    concise description of CSMA, then ref:

https://en.wikipedia.org/wiki/Carrier-sense_multiple_access

For hidden node problem, https://en.wikipedia.org/wiki/Hidden_node_problem ,

802.11n optionally (and is implemented in AREDN) uses CTS/RTS:

https://en.wikipedia.org/wiki/IEEE_802.11_RTS/CTS

...

Reply:
Sounds like a plan.  Let me rework those paragraphs to do as you suggest. ...  I'm wondering whether the "Hidden Node" and "Exposed Node" sections are even needed in the AREDN doc, or whether I should just point people to the wikipedia articles for further research.  Let me know, and I can remove those section headings, graphics, etc, if it's better that way.  THX

Reply:
How about avoiding the Hidden Node and Exposed node issues by consolidating into a single section called "Detecting a Busy Channel" -- or something like that?  Here's an example section I could use to replace both of those Node sections.  Let me know if substituting this section for the other two "node" section works better and I'll commit the changes to my side.  THX

Detecting a Busy Channel
------------------------

In any wireless network there will be nodes which are not within radio range of each other. In the example below, **A** can hear **B** but cannot hear **C**. Since **A** and **C** are hidden from each other, they may try to transmit on the same channel at the same time without knowing it. *Collision Detection* mechanisms will not help because the nodes have no way to communicate except through node **B**, so *Collision Avoidance* mechanisms must be used instead.

.. image:: _images/hidden-node.png
   :alt: Hidden Node Problem
   :align: center

AREDN |trade| firmware follows `IEEE 802.11a/b/g/n standards <https://en.wikipedia.org/wiki/IEEE_802.11n-2009>`_ and uses `Carrier Sense Multiple Access / Collision Avoidance (CSMA/CA) <https://en.wikipedia.org/wiki/Carrier-sense_multiple_access>`_ to determine whether a channel is busy.  Optionally AREDN |trade| firmware also employs `Request to Send / Clear to Send (RTS/CTS) <https://en.wikipedia.org/wiki/IEEE_802.11_RTS/CTS>`_ messages to negotiate the use of a channel. Decreased network throughput is one side effect, because several nodes may be negotiating or waiting to use the channel.

For `hidden nodes <https://en.wikipedia.org/wiki/Hidden_node_problem>`_, two approaches may help to minimize this issue. You may be able to make the hidden nodes visible to each other, for example by increasing their signal strength. The alternative is to isolate the nodes completely by placing them onto different bands or channels. Since nodes using directional antennas are nearly invisible to others not positioned in the antenna's beam, directional antennas should be used with care when sharing a channel. It may be more appropriate to create a separate link between the sites and to put the radios on a different band or channel.

Reply:
Committed the replacement to channel_planning.rst so reviewers can see the updates.  THX

Reply:
From the perspective of "what are the problems groups want to avoid" to focus in a deployment guide, it would help groups avoid scaling issues, if they didn't do some things.  For example, if there are 2 tower/cell coverage areas next to each other (or 2 in  hearing distance), they should try to use different channels.  This is what we want to state.    The reasons get into hidden/exposed node, overhead to share the channel with CSMA, etc.  Which introduces the RTS/CTS protocol overhead.   But the biggest factor is simply sharing the same channel too much.  Can reference these low level protocols but it would miss the target to focus on them.   

A simple example of a 3 hop link from QTH1-tower1-tower2-QTH2.  if all 3 links are on the same channel.   then only one link half-duplex can be sending data at a time.  It instantly cut the traffic down by 1/3 and more and high latency with protocol overhead.   Compared to each link on a different channel where each link is transmitting simultaneously.     Showing these 2 examples is what groups should understand, not the details of how RTS/CTS works or CSMA.   With same channel you can get 1 HD video and a voip call with some drop outs here and there (large range of latency).  With different channels you can do 3 HD videos, multiple voip calls and rare drop outs.

Reply:
Working to clarify now...   THX

Reply:
Added Joe's wording and example to the Channel Planning section.  Also created a sidebar to call out the main point.  Hopefully this makes it clearer.